examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+        if (frequency_idx == 0)
+          {
+            std::cout << "   Number of active cells :       "
+                      << triangulation.n_active_cells() << std::endl;
+            std::cout << "   Number of degrees of freedom : "
+                      << dof_handler.n_dofs() << std::endl;
+          }
+
+        if (frequency_idx == 0)
+          {
+            // Write the simulation parameters only once
+            parameters.data.set_attribute("active_cells",
+                                          triangulation.n_active_cells());
+            parameters.data.set_attribute("degrees_of_freedom",
+                                          dof_handler.n_dofs());
+          }
+
+        // We calculate the frequency and omega values for this particular step.
+        double current_loop_frequency =
+          (parameters.start_frequency +
+           frequency_idx *
+             (parameters.stop_frequency - parameters.start_frequency) /
+             (parameters.nb_frequency_points - 1));
+        double current_loop_omega = 2 * numbers::PI * current_loop_frequency;
+
+        // In the first frequency step we calculate the mass and stiffness
+        // matrices and the right hand side. In the subsequent frequency steps
+        // we will use those values. This improves considerably the calculation
+        // time.
+        assemble_system(current_loop_omega,
+                        (frequency_idx == 0) ? true : false);
+        solve();
+
+        frequency[frequency_idx] = current_loop_frequency;
+        store_frequency_step_data(frequency_idx);
+
+        computing_timer.print_summary();
+        computing_timer.reset();
+        pcout << std::endl;
+      }
+  }
+
+
+
+  // @sect4{ElasticWave::run}
+
+  // This function is very similar to the one in step-40.
+  template <int dim>
+  void ElasticWave<dim>::run()
+  {
+#ifdef DEBUG
+    std::cout << "Debug mode" << std::endl;
+#else
+    std::cout << "Release mode" << std::endl;
+#endif
+
+    {
+      Point<dim> p1;
+      p1(0) = -parameters.dimension_x / 2;
+      p1(1) = -parameters.dimension_y / 2;
+      if (dim == 3)
+        {
+          p1(2) = -parameters.dimension_y / 2;
+        }
+      Point<dim> p2;
+      p2(0) = parameters.dimension_x / 2;
+      p2(1) = parameters.dimension_y / 2;
+      if (dim == 3)
+        {
+          p2(2) = parameters.dimension_y / 2;
+        }
+      std::vector<unsigned int> divisions(dim);
+      divisions[0] = int(parameters.dimension_x / parameters.dimension_y);
+      divisions[1] = 1;
+      if (dim == 3)
+        {
+          divisions[2] = 1;
+        }
+      GridGenerator::subdivided_hyper_rectangle(triangulation,
+                                                divisions,
+                                                p1,
+                                                p2);
+    }
+
+    triangulation.refine_global(parameters.grid_level);
+
+    setup_quadrature_cache();
+
+    set_position_vector();
```

Comment:
I've commented above that maybe a better name for this function would be `initialize_position_vector()`. An alternative, that matches the previous call here, would be `setup_position_vector()`.